File: game/player.py
# player.py

import random

class Player:
    def __init__(self, name, is_human=True):
        self.name = name
        self.is_human = is_human
        self.promille = 0
        self.memory = 0
        self.hand = []  # Spillerens hånd med kort
        self.npcs = []  # NPC-er spilleren har hentet
        self.last_drink_card = None
        self.last_card_promille = 0
        self.status = "active"  # active, outside, standby, etc.

    def add_promille(self, amount):
        self.promille += amount
        if self.promille < 0:
            self.promille = 0
        # Promille kan ikke overstige 5
        if self.promille > 5:
            self.promille = 5

    def add_memory(self, amount):
        self.memory += amount

    def add_npc(self, npc):
        """Legg til en NPC til spilleren"""
        self.npcs.append(npc)
        print(f"{self.name} har nå {len(self.npcs)} NPC-er")

    def get_hand_size(self):
        """Hent antall kort på hånden"""
        return len(self.hand)
    
    def get_dice_bonus(self):
        """Hent total terningbonus basert på spilleren sin tilstand"""
        bonus = 0
        
        # Sweetspot bonus (1-3 promille)
        if 1 <= self.promille <= 3:
            bonus += 1
            
        # Her kan vi legge til flere bonuser senere:
        # - NPC-effekter som gir terningbonus
        # - Kort-effekter som gir terningbonus
        # - Sted-effekter som gir terningbonus
        
        return bonus

File: web/js/game/player.js
// SMØR - Player-klasse (konvertert fra game/player.py)

class Player {
  constructor(name, isHuman = true) {
    this.name = name;
    this.isHuman = isHuman;
    this.promille = 0;
    this.memory = 0;
    this.hand = []; // Spillerens hånd med kort
    this.npcs = []; // NPC-er spilleren har hentet
    this.lastDrinkCard = null;
    this.lastCardPromille = 0;
    this.status = "active"; // active, outside, standby, etc.
    this.diceBonus = 0; // Temporary dice bonus/penalty
    this.hasPill = false; // Whether player has pill penalty
  }

  addPromille(amount) {
    this.promille += amount;
    if (this.promille < 0) {
      this.promille = 0;
    }
    // Promille kan ikke overstige 5
    if (this.promille > 5) {
      this.promille = 5;
    }
  }

  addMemory(amount) {
    this.memory += amount;
  }

  addDiceBonus(amount) {
    // Add temporary dice bonus (can be negative for penalties)
    this.diceBonus += amount;
  }

  addNpc(npc) {
    // Legg til en NPC til spilleren
    this.npcs.push(npc);
  }

  getHandSize() {
    // Hent antall kort på hånden
    return this.hand.length;
  }

  getDiceBonus() {
    // Hent total terningbonus basert på spilleren sin tilstand
    let bonus = 0;

    // Sweetspot bonus (1-3 promille)
    if (this.promille >= 1 && this.promille <= 3) {
      bonus += 1;
    }

    // Temporary dice bonus (from pill, leader jersey, etc.)
    if (this.diceBonus) {
      bonus += this.diceBonus;
    }

    // Her kan vi legge til flere bonuser senere:
    // - NPC-effekter som gir terningbonus
    // - Kort-effekter som gir terningbonus
    // - Sted-effekter som gir terningbonus

    return bonus;
  }
}
